draft-ietf-quic-transport.md

```diff
@@ -3838,143 +3838,136 @@ The details of loss detection and congestion control are described in
+either the ECT(0) or ECT(1) codepoint, ECN validation fails if the
+corresponding ECN counts are not present in the ACK frame. This check detects a
+network element that zeroes the ECN field or a peer that is unable to access
+ECN the markings.
 
-ECN validation fails if the sum of the increase in ECT(0) and ECN-CE counts is
-less than the number of newly acknowledged packets that were originally sent
+ECN validation also fails if the sum of the increase in ECT(0) and ECN-CE counts
+is less than the number of newly acknowledged packets that were originally sent
 with an ECT(0) marking.  Similarly, ECN validation fails if the sum of the
 increases to ECT(1) and ECN-CE counts is less than the number of newly
 acknowledged packets sent with an ECT(1) marking.  These checks can detect
-removal of ECN markings in the network.
+remarking of ECN-CE markings by the network.
 
 An endpoint could miss acknowledgements for a packet when ACK frames are lost.
 It is therefore possible for the total increase in ECT(0), ECT(1), and ECN-CE
-counts to be greater than the number of packets acknowledged in an ACK frame.
-This is why counts are permitted to be larger than might be accounted for by
-newly acknowledged packets.
+counts to be greater than the number of packets that are newly acknowledged by
+an ACK frame. This is why ECN counts are permitted to be larger than the value
+corresponding to the largest acknowledged packet number.
 
-ECN validation MAY fail if the total count for an ECT(0) or ECT(1) marking
-exceeds the total number of packets sent with the corresponding marking.   In
-particular, an endpoint that never applies a particular marking can fail
-validation when a non-zero count for the corresponding marking is received.
-This check can detect when packets are marked ECT(0) or ECT(1) in the network.
+Validating ECN counts from reordered ACK frames can result in failure. An
+endpoint MUST NOT fail ECN validation as a result of processing an ACK frame
+that does not increase the largest acknowledged packet number.
 
-Processing ECN counts out of order can result in validation failure.  An
-endpoint SHOULD skip ECN validation on an ACK frame that does not increase the
-largest acknowledged packet number.
+ECN validation can fail if the received total count for either ECT(0) or ECT(1)
+exceeds the total number of packets sent with each corresponding ECT codepoint.
+In particular, validation will fail when an endpoint receives a non-zero ECN
+count corresponding to an ECT codepoint that it never applied.  This check
+detects when packets are remarked to ECT(0) or ECT(1) in the network.
 
 
-#### Validation Outcomes
+#### ECN Validation Outcomes
 
-If validation fails, then the endpoint stops sending ECN markings in subsequent
-IP packets with the expectation that either the network path or the peer does
-not support ECN.
+If validation fails, then the endpoint MUST disable ECN. It stops setting the
```

Comment:
I don't think this should be a MUST. If normative language is needed I guess it should be a SHOULD with a warning that remarking by the network can conceal congestion. I would however rather simply not use normative language here.

Reply:
I think it is a MUST, until the next round of validation starts. The endpoint shouldn't continue to set ECT(?) and not be able to respond when CE is marked.

Reply:
This is what the original text said, so I think the intent is that of a MUST.